`dotnet publish` fails on macOS on ARM

Hi,

I tried to publish an image from a fresh web app, but hit an `ObjectDisposedException`:

```text
❯ dotnet new web -n test-container    
The template "ASP.NET Core Empty" was created successfully.

Processing post-creation actions...
Restoring /Users/dkBjMaMa/test-container/test-container.csproj:
  Determining projects to restore...
  Restored /Users/dkBjMaMa/test-container/test-container.csproj (in 30 ms).
Restore succeeded.

❯ cd test-container

❯ dotnet add package Microsoft.NET.Build.Containers             
...

❯ dotnet publish --os linux --arch x64 -c Release -p:PublishProfile=DefaultContainer
MSBuild version 17.4.0+18d5aef85 for .NET
  Determining projects to restore...
  Restored /Users/User/test-container/test-container.csproj (in 83 ms).
  test-container -> /Users/User/test-container/bin/Release/net7.0/linux-x64/test-container.dll
  test-container -> /Users/User/test-container/bin/Release/net7.0/linux-x64/publish/
  Building image 'test-container' with tags 1.0.0 on top of base image mcr.microsoft.com/dotnet/aspnet:7.0
/Users/User/.nuget/packages/microsoft.net.build.containers/0.2.7/build/Microsoft.NET.Build.Containers.targets(124,9): error MSB4018: The "CreateNewImage" task failed unexpectedly. [/Users/User/test-container/test-container.csproj]
/Users/User/.nuget/packages/microsoft.net.build.containers/0.2.7/build/Microsoft.NET.Build.Containers.targets(124,9): error MSB4018: System.ObjectDisposedException: Cannot access a disposed object. [/Users/User/test-container/test-container.csproj]
/Users/User/.nuget/packages/microsoft.net.build.containers/0.2.7/build/Microsoft.NET.Build.Containers.targets(124,9): error MSB4018: Object name: 'System.Security.Cryptography.SHA256+Implementation'. [/Users/User/test-container/test-container.csproj]
/Users/User/.nuget/packages/microsoft.net.build.containers/0.2.7/build/Microsoft.NET.Build.Containers.targets(124,9): error MSB4018:    at System.Security.Cryptography.HashAlgorithm.TransformFinalBlock(Byte[] inputBuffer, Int32 inputOffset, Int32 inputCount) [/Users/User/test-container/test-container.csproj]
/Users/User/.nuget/packages/microsoft.net.build.containers/0.2.7/build/Microsoft.NET.Build.Containers.targets(124,9): error MSB4018:    at System.Security.Cryptography.CryptoStream.FlushFinalBlockAsync(Boolean useAsync, CancellationToken cancellationToken) [/Users/User/test-container/test-container.csproj]
/Users/User/.nuget/packages/microsoft.net.build.containers/0.2.7/build/Microsoft.NET.Build.Containers.targets(124,9): error MSB4018:    at System.Security.Cryptography.CryptoStream.FlushFinalBlock() [/Users/User/test-container/test-container.csproj]
/Users/User/.nuget/packages/microsoft.net.build.containers/0.2.7/build/Microsoft.NET.Build.Containers.targets(124,9): error MSB4018:    at System.Security.Cryptography.CryptoStream.Dispose(Boolean disposing) [/Users/User/test-container/test-container.csproj]
/Users/User/.nuget/packages/microsoft.net.build.containers/0.2.7/build/Microsoft.NET.Build.Containers.targets(124,9): error MSB4018:    at System.IO.Stream.Close() [/Users/User/test-container/test-container.csproj]
/Users/User/.nuget/packages/microsoft.net.build.containers/0.2.7/build/Microsoft.NET.Build.Containers.targets(124,9): error MSB4018:    at Microsoft.NET.Build.Containers.Layer.HashDigestGZipStream.Dispose(Boolean disposing) in D:\a\_work\1\s\Microsoft.NET.Build.Containers\Layer.cs:line 143 [/Users/User/test-container/test-container.csproj]
/Users/User/.nuget/packages/microsoft.net.build.containers/0.2.7/build/Microsoft.NET.Build.Containers.targets(124,9): error MSB4018:    at Microsoft.NET.Build.Containers.Layer.FromFiles(IEnumerable`1 fileList) in D:\a\_work\1\s\Microsoft.NET.Build.Containers\Layer.cs:line 59 [/Users/User/test-container/test-container.csproj]
/Users/User/.nuget/packages/microsoft.net.build.containers/0.2.7/build/Microsoft.NET.Build.Containers.targets(124,9): error MSB4018:    at Microsoft.NET.Build.Containers.Layer.FromDirectory(String directory, String containerPath) in D:\a\_work\1\s\Microsoft.NET.Build.Containers\Layer.cs:line 24 [/Users/User/test-container/test-container.csproj]
/Users/User/.nuget/packages/microsoft.net.build.containers/0.2.7/build/Microsoft.NET.Build.Containers.targets(124,9): error MSB4018:    at Microsoft.NET.Build.Containers.Tasks.CreateNewImage.Execute() in D:\a\_work\1\s\Microsoft.NET.Build.Containers\CreateNewImage.cs:line 212 [/Users/User/test-container/test-container.csproj]
/Users/User/.nuget/packages/microsoft.net.build.containers/0.2.7/build/Microsoft.NET.Build.Containers.targets(124,9): error MSB4018:    at Microsoft.Build.BackEnd.TaskExecutionHost.Microsoft.Build.BackEnd.ITaskExecutionHost.Execute() [/Users/User/test-container/test-container.csproj]
/Users/User/.nuget/packages/microsoft.net.build.containers/0.2.7/build/Microsoft.NET.Build.Containers.targets(124,9): error MSB4018:    at Microsoft.Build.BackEnd.TaskBuilder.ExecuteInstantiatedTask(ITaskExecutionHost taskExecutionHost, TaskLoggingContext taskLoggingContext, TaskHost taskHost, ItemBucket bucket, TaskExecutionMode howToExecuteTask) [/Users/User/test-container/test-container.csproj]
```

My system:
```text
System Version: macOS 13.0 (22A380)
Model Identifier: MacBookPro18,1
Chip: Apple M1 Pro

❯ dotnet --info                                                                     
.NET SDK:
 Version:   7.0.100
 Commit:    e12b7af219

Runtime Environment:
 OS Name:     Mac OS X
 OS Version:  13.0
 OS Platform: Darwin
 RID:         osx.13-arm64
 Base Path:   /usr/local/share/dotnet/sdk/7.0.100/

Host:
  Version:      7.0.0
  Architecture: arm64
  Commit:       d099f075e4

.NET SDKs installed:
  6.0.402 [/usr/local/share/dotnet/sdk]
  7.0.100 [/usr/local/share/dotnet/sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.App 6.0.10 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 7.0.0 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 6.0.10 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 7.0.0 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]

Other architectures found:
  None

Environment variables:
  Not set

global.json file:
  Not found

Learn more:
  https://aka.ms/dotnet/info

Download .NET:
  https://aka.ms/dotnet/download
```

--- a/Microsoft.NET.Build.Containers/Layer.cs
+++ b/Microsoft.NET.Build.Containers/Layer.cs
@@ -139,8 +139,10 @@ public record struct Layer
         {
             try
             {
-                hashAlgorithm.Dispose();
+                // dispose hashAlgorithm after sha256Stream since sha256Stream references/uses it
                 sha256Stream.Dispose();
+                hashAlgorithm.Dispose();
+
                 compressionStream.Dispose();
             }
             finally

